packages/protocol/scripts/foundry/make-release.ts

```diff
@@ -0,0 +1,805 @@
+
+      try {
+        const content = readJsonSync(artifactFilePath)
+        if (content.abi && content.bytecode) {
+          contractArtifactPathsMap.set(contractName, artifactFilePath)
+        } else {
+          console.warn(`Skipping non-ABI/bytecode file: ${artifactFilePath}`)
+        }
+      } catch (e) {
+        console.warn(`Skipping non-JSON or unreadable file: ${artifactFilePath}`)
+      }
+    }
+  }
+}
+
+const linkLibraries = (
+  contractViemArtifact: ViemContract,
+  contractDependencies: string[],
+  addresses: ContractAddresses
+): void => {
+  if (!contractViemArtifact.bytecode.includes('__')) {
+    return
+  }
+
+  if (contractDependencies.length === 0) {
+    console.error(
+      `No dependencies found for ${contractViemArtifact.contractName}. Skipping library linking.`
+    )
+    return
+  }
+
+  for (const dep of contractDependencies) {
+    if (addresses.addresses.has(dep)) {
+      const libAddress = addresses.get(dep).replace('0x', '')
+      const libSourceFilePath = contractViemArtifact.sourceFiles.find((file) =>
+        file.includes(`${dep}.sol`)
+      )
+
+      if (!libSourceFilePath) {
+        console.error(
+          `Could not determine sourceFilePath for library ${dep}. Skipping Foundry placeholder replacement for it in ${contractViemArtifact.contractName}.`
+        )
+        return
+      }
+
+      const stringToHash = `${libSourceFilePath}:${dep}`
+      const hashed = keccak256(toHex(new TextEncoder().encode(stringToHash)))
+      const placeholderHash = hashed.substring(2, 2 + 34)
+
+      const placeholderRegexDollar = new RegExp(`__\\$${placeholderHash}\\$__`, 'g')
+      if (contractViemArtifact.bytecode!.match(placeholderRegexDollar)) {
+        contractViemArtifact.bytecode = contractViemArtifact.bytecode!.replace(
+          placeholderRegexDollar,
+          libAddress
+        ) as Hex
+      } else {
+        console.log(`No placeholder match for ${dep} in ${contractViemArtifact.contractName}.`)
+      }
+    }
+  }
+}
+
+const performRelease = async (
+  contractName: string,
+  report: ASTDetailedVersionedReport,
+  released: Set<string>,
+  contractArtifactPaths: Map<string, string>,
+  dependencies: { get(key: string): string[] | undefined },
+  addresses: ContractAddresses,
+  proposal: ProposalTx[],
+  initializationData: Record<string, unknown[]>,
+  walletClient: WalletClient,
+  publicClient: PublicClient
+): Promise<void> => {
+  if (released.has(contractName)) return
+
+  const shouldDeployContract = Object.keys(report.contracts).includes(contractName)
+  const shouldDeployLibrary = Object.keys(report.libraries).includes(contractName)
+
+  if (!shouldDeployContract && !shouldDeployLibrary) return
+
+  const artifactPath = contractArtifactPaths.get(contractName)
+  if (!artifactPath) {
+    console.error(`Artifact path for ${contractName} not found in map. Skipping.`)
+    released.add(contractName)
+    return
+  }
```

Comment:
Wondering if marking such contract as released won't break anything other that depends on it 🤔
Maybe better would be to throw error catched outside in `main()` function?